cpp/src/slice2cpp/Gen.cpp

```diff
@@ -1284,7 +1288,7 @@ Slice::Gen::ForwardDeclVisitor::visitEnum(const EnumPtr& p)
     {
         H << "class ";
     }
-    H << fixKwd(p->name());
+    H << getDeprecatedSymbol(p) << fixKwd(p->name());
```

Comment:
I think it's preferable that these kinds of functions do return strings with padding. I think it's cleaner to emit the deprecation in an if check on the line above.

Reply:
Unfortunately there's no way to get a function like `emitDeprecationAttribute(H, p)` working, because the function can't know whether to put the space in front, or in back, of the thing (which we want depends on the context).

But, the logic for generating this space also _has_ to be in this function, because an outside caller won't know if the attribute was actually written or not, and so it can't know whether or not to put a space.

----

I'll still try looking into simplifying it, but it probably won't be as clean an approach as suggested.